### PR TITLE
prepare release 1.6.31

### PR DESCRIPTION
**Pull request by vvoland**
- update containerd binary to v1.6.31
- Update Golang runtime to 1.21.9

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,9 @@
+containerd.io (1.6.31-2) release; urgency=medium
+
+  * update containerd binary to v1.6.31.
+
+ -- Paweł Gronowski <pawel.gronowski@docker.com>  Fri, 08 Apr 2024 13:31:20 +0000
```
Comment (vvoland) [inline debian/changelog:5]: Should I adjust the date? 🤔

**Review by laurazard — DISMISSED** [review]

File: debian/changelog (hunk 1 above)
Comment (laurazard) [reply debian/changelog:5]: I guess? The date matches the commit though 👀

File: debian/changelog [hunk 2]
```diff
@@ -1,3 +1,9 @@
+containerd.io (1.6.31-2) release; urgency=medium
+
+  * update containerd binary to v1.6.31.
```
Comment (thaJeztah) [inline debian/changelog:3]: We should build with latest go1.21 version (upstream didn't update yet, but if we build our packages, we can override the version)

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Pull request merged**